In [2]:
import pandas as pd

In [4]:
import pandas as pandas
path = "GlobalWeatherReposiroty.csv"
df = pd.read_csv(path)
df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


In [5]:
df.head()

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.52,69.18,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.4,26.6,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.33,19.82,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.1,2.0,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.76,3.05,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.4,18.4,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.50,1.52,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.7,0.9,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.84,13.23,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.4,262.3,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55


In [6]:
df.tail()

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22
39593,Zimbabwe,Harare,-17.8178,31.0447,Africa/Harare,1733482800,12/6/2024 13:00,34.7,94.4,Sunny,...,12.950,13.505,1,2,5:13 AM,6:21 PM,9:43 AM,10:58 PM,Waxing Crescent,22


In [9]:
df_noid = df.iloc[:,7]
df_noid

0        26.6
1        19.0
2        23.0
3         6.3
4        26.0
         ... 
39589    16.2
39590    21.2
39591    22.2
39592    36.0
39593    34.7
Name: temperature_celsius, Length: 39594, dtype: float64

In [10]:
df_noid.describe()

count    39594.000000
mean        24.350599
std          8.130323
min        -21.900000
25%         20.200000
50%         26.100000
75%         29.200000
max         49.200000
Name: temperature_celsius, dtype: float64

In [24]:
df_noid[['pressure_mb','humidity']].mean()

pressure_mb    1013.195914
humidity         62.528237
dtype: float64

In [25]:
df_noid[['temperature_celsius', 'humidity','air_quality_Ozone']].corr()

,temperature_celsius,humidity,air_quality_Ozone
temperature_celsius,1.000000,-0.420825,0.334880
humidity,-0.420825,1.000000,-0.508499
air_quality_Ozone,0.334880,-0.508499,1.000000


Tugas 1 Week 2_42324043 Sophia Melisa Silitonga

Nomor 1


In [26]:
# Analisis korelasi Tugas 1
df_noid[['temperature_fahrenheit',
         'pressure_mb',
         'air_quality_Sulphur_dioxide']].corr()

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
temperature_fahrenheit,1.000000,-0.486833,-0.023152
pressure_mb,-0.486833,1.000000,0.018620
air_quality_Sulphur_dioxide,-0.023152,0.018620,1.000000


Hasil Analisis No.1

1.Temperature Fahrenheit dan Pressure mb = -0.486833,
Menunjukkan hubungan negatif sedang. Saat suhu meningkat, tekanan udara cenderung menurun.

2.Temperature Fahrenheit dan Sulphur dioxide = -0.023152,
Menunjukkan hubungan sangat lemah atau hampir tidak ada.

3.Pressure mb dan Sulphur dioxide = 0.018620,
Menunjukkan hubungan positif yang sangat lemah atau hampir tidak ada.

Kesimpulan:
Hubungan paling terlihat adalah antara temperature Fahrenheit dan pressure mb dengan korelasi -0.486833. Dua pasangan lainnya memiliki korelasi yang sangat lemah karena nilain

Nomor 2A Analisa mean, std, min, max, Q1, Q2, Q3 untuk data Previous Purchases
(Dengan DATASET shopping_trends.csv)

In [31]:
import pandas as pd

df_customer = pd.read_csv("shopping_trends.csv")

df_customer.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually


In [38]:
df_customer['Previous Purchases'].describe()[['mean', 'std', 'min', 'max', '25%', '50%', '75%']]

mean    25.351538
std     14.447125
min      1.000000
max     50.000000
25%     13.000000
50%     25.000000
75%     38.000000
Name: Previous Purchases, dtype: float64

Hasil Analisis:

Rata-rata jumlah pembelian sebelumnya adalah 25,35 kali. Nilai terendah adalah 1 kali dan tertinggi 50 kali. 
Sebanyak 50% data berada pada rentang 13–38 pembelian sebelumnya.

Nomor 2B Analisa korelasi Age vs Purchase Amount (USD) dan Purchase Amount (USD) vs Review Rating
(Dengan DATASET shopping_trends.csv)

1.Korelasi Age vs Purchase Amount (USD)

In [39]:
df_customer[['Age', 'Purchase Amount (USD)']].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


Hasil Analisis:    Age vs Purchase Amount (USD) = -0,010424
Nilainya sangat mendekati 0, sehingga hubungan antara usia dan jumlah pembelian sangat lemah atau hampir tidak ada.

2. Korelasi Purchase Amount (USD) vs Review Rating

In [40]:
df_customer[['Purchase Amount (USD)', 'Review Rating']].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


Hasil Analisis :Purchase Amount (USD) vs Review Rating = 0,030776
Nilainya juga sangat mendekati 0, sehingga hubungan antara jumlah pembelian dan rating ulasan sangat lemah atau hampir tidak ada.

KESIMPULAN: Kedua variabel tersebut memiliki hubungan linear yang sangat lemah, karena nilai korelasinya mendekati 0.